include/vcpkg/base/messages.h

```diff
@@ -232,7 +282,15 @@ namespace vcpkg::msg
     constexpr static struct NAME##_t                                                                                   \
     {                                                                                                                  \
         constexpr static const char* name = #NAME;                                                                     \
-        constexpr static const char* example = EXAMPLE;                                                                \
+        constexpr static StringArray example = EXAMPLE;                                                                \
+        constexpr static StringArray example_str = "An example of {" #NAME "} is " EXAMPLE ".";                        \
+        constexpr static auto real_example()                                                                           \
```

Comment:
If no example is provided, return the empty example, otherwise return the already formatted example string. I don't think we need the member `example` of this struct anymore.